tests/unit/invalid-document-test.js

```diff
@@ -434,43 +434,6 @@ module('Unit | Document', function(hooks) {
     });
   });
 
```

Comment:
BTW validation for links is "slightly" different for each of the three things I mentioned, here are the relevant spec links:

### Document Links

Document links have `self` with is a `string` or a `links object` (`href` + optional `meta`).
They appear at the same level as the primary document (e.g. alongside the `data` `included` `meta` members). Documents that contain collections (`data` is an array) may also have `pagination` links.

- http://jsonapi.org/format/#document-links
- http://jsonapi.org/format/#fetching-pagination

example:

```
{
  data: [],
  links: { self: 'https://api.example.com/foos' }
}
```


### Resource Links

Resource links have the same rules as `document` links, and appear at the top level of the resource. e.g. alongside `type` `id` `attributes` and `relationships`. **Resources cannot have pagination links.**

example:

```
{
  data: {
    type: 'foo',
    id: '1',
    attributes: {},
    links: { self: 'https://api.example.com/foos/1' }
  }
}
```

### Relationship Links

Relationship links appear within the `Relationship` Object. **They have more requirements than `document` and `resource` links.** The `related` link is the most important aspect that is different. Relationships that represent collections may have pagination links. In `ember-data` we currently *only* utilize the `related` link. This is actually a bug, because we should fall back to the `self` link when `related` is not present and error if `links` is present without either.

- http://jsonapi.org/format/#document-resource-object-relationships
- http://jsonapi.org/format/#fetching-pagination

```
{
  data: {
    type: 'foo',
    id: '1',
    relationships: {
       bars: {
         data: [],
         links: {
           related: 'https://api.example.com/foos/1/bars'
         }
       }
    }
  }
}
```